collective/deletepermission/del_object_dexterity.py

```diff
@@ -0,0 +1,22 @@
+from AccessControl import Unauthorized
+from AccessControl import getSecurityManager
+from plone.dexterity.content import Container
+
+
+def manage_delObjects(self, ids=None, REQUEST=None):
+    """We need to enforce security."""
+    sm = getSecurityManager()
+    if not sm.checkPermission('Delete objects', self):
+        raise Unauthorized(
+            "Do not have permissions to remove this object")
+
+    if ids is None:
+        ids = []
+    if isinstance(ids, basestring):
+        ids = [ids]
+    for id_ in ids:
+        item = self._getOb(id_)
+        if not sm.checkPermission("Delete portal content", item):
+            raise Unauthorized(
+                "Do not have permissions to remove this object")
+    return super(Container, self).manage_delObjects(ids, REQUEST=REQUEST)
```

Comment:
Since this is a copy of the `manage_delObjects` in `del_object.py` except for the last line:
could you try to extract and use a separate function in `del_object.py`, e.g.

```python
# del_object.py
def protect_manage_delObjects(context, ids=None):
    sm = getSecurityManager()
    ....
            raise Unauthorized(
                "Do not have permissions to remove this object")

def manage_delObjects(self, ids=None, REQUEST=None):
    """We need to enforce security."""
    protect_manage_delObjects(self, ids)
    return PortalFolder.manage_delObjects(self, ids, REQUEST=REQUEST)
```

Then you could simply reuse it for dexterity:
```python
# del_object_dexterity.py
from collective.deletepermission.del_object import protect_manage_delObjects

def manage_delObjects(self, ids=None, REQUEST=None):
    """We need to enforce security."""
    protect_manage_delObjects(self, ids)
    return super(Container, self).manage_delObjects(ids, REQUEST=REQUEST)
```

This would reduce the duplicity and therefore make maintaining the code easier. :wink: